### ISSUE TITLE
Check math and PGM diagrams are correct and consistent

### ISSUE THREAD

**Issue by kponder** [kbarbary/snpgm#10]
For completeness! Let's assign this to @drphilmarshall :-)

**Comment by drphilmarshall** [comment]
OK, I have made some progress, in the master branch of [my fork](https://github.com/drphilmarshall/snpgm). I got halfway through @rbiswas4 's math, made some tweaks to try and explain what I think it meant in terms of the PGMs, and then also made a couple new PGMs. I suggest we pass this to @jmeyers314 for his comments, and perhaps some more math on the equations implied by these PGMs. Anyway, for your amusement, here are my new PGMs:

![image](https://cloud.githubusercontent.com/assets/710903/6745024/2dbacc50-ce6e-11e4-92c9-acf40e909d15.png)

![image](https://cloud.githubusercontent.com/assets/710903/6745029/35a2fae6-ce6e-11e4-9ca7-c611ce4e2251.png)

**Comment by kbarbary** [comment]
I might come down to stanford or slac tomorrow. Will any of you be around?

**Comment by drphilmarshall** [comment]
Yep, I'll be here - I expect @kponder and @jmeyers314 will be as well. Its KIPAC tea at 10:30am, you should aim to be here for 10am coffee! Great stuff, looking fwd to seeing you.

**Comment by wmwv** [comment]
@kbarbary That would be great.  I'll be around all day.

**Comment by kbarbary** [comment]
Should we hold off on merging your master branch, @drphilmarshall?

At first glance looks fine but I haven't looked over the math closely. The `pip-reqirements.txt` file should be removed at some point though; not sure why @rbiswas4 added that...

**Comment by drphilmarshall** [comment]
Yeah, let's write out the importance sampling concept and formula in such a
way that matches what @rbiswas wrote, and then I'll submit a PR.

**Comment by jmeyers314** [comment]
![img_0888](https://cloud.githubusercontent.com/assets/3650485/6792676/9462a03e-d177-11e4-8328-b5e4a24d583f.jpg)

Here's my attempt for including calibration, based off of Figure 1 from March++11 here: http://arxiv.org/pdf/1102.3237v4.pdf

I followed the example in this paper and included a number of deterministic relationships that, though not strictly needed since they integrate out trivially, I think help to better illustrate the relationships between variables.  Note that deterministic conditional PDFs (i.e., delta functions) are indicated with orange arrows, and probabilistic ones indicated in blue.

The plates (boxes) separate variables that are indexed by SN i, band j, and epoch k.  I left this a bit sloppy: e.g., if the 4th SN isn't observed in the 2nd epoch of the 6th band, then that entry in the likelihood function should just be 1.

I think this also illustrates why importance sampling is so hard for the calibration variables: they form a deterministic relationship with the variables in the interim samples, which means that when you're reweighting the samples in the outer loop of importance sampling, the weight is a ratio of delta functions with (generally) non-overlapping support.

I'll happily code this PGM up in daft, but I'll wait to hear if I made any errors first.  Thanks!

**Comment by drphilmarshall** [comment]
In standard PGM style the circles with all orange arrows pointing at them
would be represented by dots (for delta function PDFs) I think - but I am
all in favour of colored graphs! Daft enable these very straightforwardly.

We realized today that in the March work, the SALT outputs re assumed to be
treatable with a multivariate Gaussian, such that under other, matching,
Gaussian assumptions, you can marginalized out the individual SN parameters
analytically. Working thi through in the calibration uncertainty model
would make for a happy weekend of Gaussian integrals :-)

An even more interesting generalization would be to replace all of March's
single Gaussians with Gaussian mixtures - but first thing first I guess.

I still think its worth trying to marginalize the calibration uncertainty
joint PDF with importance sampling!
